python/cuspatial/cuspatial/core/gis.py

```diff
@@ -62,12 +65,20 @@ def directed_hausdorff_distance(x, y, count):
     Returns
     -------
     DataFrame: The pairwise directed distance matrix with one row and one
-    column per input trajectory; the value at row i, column j represents the
-    hausdorff distance from trajectory i to trajectory j.
+    column per input space; the value at row i, column j represents the
+    hausdorff distance from space i to space j.
     """
-    result = cpp_directed_hausdorff_distance(x, y, count)
-    dim = len(count)
-    return DataFrame.from_gpu_matrix(result.to_gpu_array().reshape(dim, dim))
+    num_spaces = len(points_per_space)
+    if num_spaces == 0:
+        return DataFrame()
+    result = cpp_directed_hausdorff_distance(
+        xs.astype("float64")._column,
+        ys.astype("float64")._column,
+        points_per_space.astype("int32")._column,
+    )
+    result = result.to_gpu_array()
```

Comment:
Is the C++ API not outputting in the format expected? Is this just zero-copy reshaping?

Reply:
the cpp deals in columns and returns a flattened matrix as a column for perf reasons (assuming), and the python layer wants it in a matrix. I've been leaving most of the cython/python code as-is, mostly because the answer to a lot of these questions is "I don't know for sure", and my primary focus has been deleting legacy. :man_shrugging: .

Reply:
`to_gpu_array()` might've changed since this was written -- it used to return a zero-copy view, but now it does `dropna()` or `fillna()`, and returns the view of the non-null result. So likely better to change this to:
```suggestion
    result = cupy.asarray(result.data_array_view)
```